app/src/main/java/ghcharactertracker/com/ghcharactertracker/ScenarioActivity.java

```
@@ -0,0 +1,231 @@
+        healthText = (EditText) findViewById(R.id.scenarioHealth);
+        expText = (EditText) findViewById(R.id.scenarioExp);
+        moneyText = (EditText) findViewById(R.id.scenarioMoney);
+
+        dbHandler = DBHandler.getDbHandler(this);
+
+        addTextWatchers();
+        updateUI();
+    }
+
+    public void changeHealth(View v) {
+        switch(v.getId()){
+            case R.id.healthAdd:
+                updateModelScenarioHealth(scenario.getHealth() + 1);
+                break;
+            case R.id.healthSub:
+                updateModelScenarioHealth(scenario.getHealth() - 1);
+                break;
+        }
+
+        updateHealthUI();
+    }
+
+    public void changeExp(View v) {
+        switch(v.getId()){
+            case R.id.expAdd:
+                updateModelScenarioExp(scenario.getExp() + 1);
+                break;
+            case R.id.expSub:
+                updateModelScenarioExp(scenario.getExp() - 1);
+                break;
+        }
+
+        updateExpUI();
+    }
+
+    public void changeMoney(View v) {
+        switch(v.getId()){
+            case R.id.moneyAdd:
+                updateModelScenarioMoney(scenario.getMoneyTokens() + 1);
+                break;
+            case R.id.moneySub:
+                updateModelScenarioMoney(scenario.getMoneyTokens() - 1);
+                break;
+        }
+
+        updateMoneyUI();
+    }
+
+    private void updateUI() {
+        updateHealthUI();
+        updateExpUI();
+        updateMoneyUI();
+        updateLevelUI();
+    }
+
+    private void updateHealthUI() {
+        healthText.setText(Integer.toString(scenario.getHealth()));
+    }
+
+    private void updateExpUI() {
+        expText.setText(Integer.toString(scenario.getExp()));
+    }
+
+    private void updateMoneyUI() {
+        moneyText.setText(Integer.toString(scenario.getMoneyTokens()));
+    }
+
+    private void updateLevelUI() {
+        if (scenario.getLevel() > 0) {
```

this nad the next couple functions can silently fail.

that ok?

---

Silently fail meaning they don't get through the if? that's fine, if the store model info and the info we're being fed from the UI is the same then there are no changes to make in the model. honestly I'm not making and performance impact with these `if` checks so I can probably remove them.

---

kk dank